src/internal.c

```diff
@@ -24793,9 +24799,12 @@ int SendCertificate(WOLFSSL* ssl)
     if (ssl->fragOffset != 0)
         length -= (ssl->fragOffset + headerSz);
 
-    maxFragment = MAX_RECORD_SIZE;
 
-    maxFragment = (word32)wolfSSL_GetMaxFragSize(ssl, (int)maxFragment);
+    maxFragment = (word32)wolfSSL_GetMaxPlaintextSize(ssl);
+    if (ssl->options.dtls)
+        maxFragment -= DTLS_HANDSHAKE_HEADER_SZ;
+    else
+        maxFragment -= HANDSHAKE_HEADER_SZ;
```

Comment:
Wouldn't it be better to have an equivalent `wolfSSL_GetMaxCiphertextSize`?

Reply:
`ciphertextsize` will be wrong naming as we are computing the maximum plaintext "payload" that can fit in a record. 
I like more to leave the function dumber and to return the maximum plaintext and move overhead adjustment (like handshake headers) to the function's users.